websocketOutportDriver/client/wsClient.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/gorilla/websocket"
+	"github.com/multiversx/mx-chain-core-go/websocketOutportDriver"
+	"github.com/multiversx/mx-chain-core-go/websocketOutportDriver/data"
+	"github.com/multiversx/mx-chain-core-go/websocketOutportDriver/sender"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+const closedConnection = "use of closed network connection"
+
+var log = logger.GetOrCreate("wsClient")
+
+type client struct {
+	url                      string
+	blockingOnError          bool
+	retryDuration            time.Duration
+	wsConn                   WSConnClient
+	payloadParser            PayloadParser
+	operationHandler         OperationHandler
+	uint64ByteSliceConverter sender.Uint64ByteSliceConverter
+}
+
+type ArgsWsClient struct {
+	Url                      string
+	RetryDurationInSec       uint32
+	BlockingAckOnError       bool
+	OperationHandler         OperationHandler
+	PayloadParser            PayloadParser
+	Uint64ByteSliceConverter sender.Uint64ByteSliceConverter
+	WSConnClient             WSConnClient
+}
+
+// NewWsClientHandler will create a ws client to receive data from an observer/light client
+func NewWsClientHandler(args *ArgsWsClient) (*client, error) {
+	if args.OperationHandler == nil {
+		return nil, errNilOperationHandler
+	}
+	if args.PayloadParser == nil {
+		return nil, errNilPayloadParser
+	}
+	if args.WSConnClient == nil {
+		return nil, errNilWsConnReceiver
+	}
+	if args.Uint64ByteSliceConverter == nil {
+		return nil, errNilUint64ByteSliceConverter
+	}
+	if len(args.Url) == 0 {
+		return nil, errEmptyUrlProvided
+	}
+
+	urlReceiveData := url.URL{Scheme: "ws", Host: args.Url, Path: data.WSRoute}
+	retryDuration := time.Duration(args.RetryDurationInSec) * time.Second
+
+	return &client{
+		url:                      urlReceiveData.String(),
+		blockingOnError:          args.BlockingAckOnError,
+		retryDuration:            retryDuration,
+		wsConn:                   args.WSConnClient,
+		payloadParser:            args.PayloadParser,
+		operationHandler:         args.OperationHandler,
+		uint64ByteSliceConverter: args.Uint64ByteSliceConverter,
+	}, nil
+}
+
+// Start will start the client listening ws process
+func (c *client) Start() {
+	log.Info("connecting to", "url", c.url)
+
+	for {
+		err := c.wsConn.OpenConnection(c.url)
+		if err != nil {
+			log.Warn(fmt.Sprintf("c.openConnection(), retrying in %v...", c.retryDuration), "error", err.Error())
+			time.Sleep(c.retryDuration)
+			continue
+		}
+
+		closed := c.listenOnWebSocket()
+		if closed {
+			return
+		}
+	}
+}
+
+func (c *client) listenOnWebSocket() (closed bool) {
+	for {
+		_, message, err := c.wsConn.ReadMessage()
+		if err == nil {
+			c.verifyPayloadAndSendAckIfNeeded(message)
+			continue
+		}
+
+		_, isConnectionClosed := err.(*websocket.CloseError)
+		if !isConnectionClosed {
+			if strings.Contains(err.Error(), closedConnection) {
+				return true
+			}
+			log.Warn("c.listenOnWebSocket()-> connection problem, retrying", "error", err.Error())
```

Comment:
return after this line and remove else statement?

Reply:
Applied suggestion